be/src/agent/task_worker_pool.cpp

```diff
@@ -943,7 +943,7 @@ void* TaskWorkerPool::_update_tablet_meta_worker_thread_callback(void* arg_this)
                     // because the primary index is available in cache
                     // But it will be remove from index cache after apply is finished
                     auto manager = StorageEngine::instance()->update_manager();
-                    manager->index_cache().remove_by_key(tablet->tablet_id());
+                    manager->index_cache().try_remove_by_key(tablet->tablet_id());
```

Comment:
If the try_remove_by key is false, when to remove the key?

Reply:
> If the try_remove_by key is false, when to remove the key?

It will be remove from index cache after apply is finished